### PR TITLE
Bug 1902029: move off docker.io in build e2e's (approximate recent 4.7 changes)

### PR DESCRIPTION
**Pull request by gabemontero**
This PR picks the move to imagestream dockerimage ref's achived via https://github.com/openshift/origin/pull/24887 in 4.7 for the build e2e's (vs. use 
of busybox, centos:7, or direct ruby etc. refs to docker.io), without the additional k8s test integration and disconnected/mirroring related changes that at the moment are only in 4.7

There is mentioned of backporting some form of that entire change to 4.6.z

But this would be a more immediate subset of that, and should not conflict if that entire backport actually occurs.

/assign @adambkaplan

### CODE REVIEW

**Comment by r4f4** [comment]
Nice! This PR has the added benefit of making it possible to run more e2e tests on s390x by removing the dependency on `centos`.

**Comment by gabemontero** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/25721/pull-ci-openshift-origin-release-4.6-e2e-gcp-builds/1333536681587904512 has some failures I need to look at ... most likely related to the change in images ... parsing for messages in build logs and such, build name refs, etc.

**Comment by gabemontero** [comment]
more cloud quota pain

/test e2e-gcp-builds

**Comment by gabemontero** [comment]
more cloud quota pain

/test e2e-gcp-builds

**Comment by gabemontero** [comment]
OK all green e2e-gcp-builds !!

/retest

**Comment by gabemontero** [comment]
fip flakes unrelated to build

/skip

**Comment by gabemontero** [comment]
bump @adambkaplan 

this 4.6.z pr along is ready for review

the bz bot will be happy once https://github.com/openshift/origin/pull/25740 merges and I can move the 4.7 bz to verified

**Comment by gabemontero** [comment]
ok pushed an analogous 4.6 change for Bug 1904683 

time to babysit e2e's again

**Comment by gabemontero** [comment]
ok e2e-gcp-builds is clean ... that is the key

e2e-gcp failures are not in sig-build

/skip

/retest

**Comment by gabemontero** [comment]
gcp throttling on e2e-gcp

**Comment by gabemontero** [comment]
more unrelated flakes in e2e-gcp

/retest

**Comment by abhat** [comment]
@eparis we need to get this merged for the e2e-vsphere-ovn job to pass on openshift/ovn-kubernetes/pull/374.

**Comment by gabemontero** [comment]
so yeah @eparis @abhat the path  to getting the bugzilla bot happy here is 
- merge 4.7 https://github.com/openshift/origin/pull/25754 to get past a newly discovered e2e timing window in 4.7 e2e builds ... that should be happening soon if we don't hit unrelated flakes
- then merge this PR's dependent bug, https://github.com/openshift/origin/pull/25745 which cannot get a passing test since last Friday cause of the first item
- once that merges and verifies, the bugzilla bot will be happy for this one

or you override the bugzilla bot

**Comment by gabemontero** [comment]
OK bugzilla bot is happy

@eparis @abhat @adambkaplan this PR is ready for lgtm approve cherrypick approve labelling if we want to merge these changes now vs. wait for pick of Clayton's recent overhaul of the e2e's to facilitate disconnected testing (which also addressed docker.io usage as well, including sig-build e2e's)

**Comment by gabemontero** [comment]
ok e2e-gcp-builds is clean once we get a cluster started ... that's the key for this change 

/assign @adambkaplan 

look like @eparis would like to get this merged @adambkaplan assuming you think this is LGTY

File: test/extended/builds/multistage.go
Comment (coreydaley) [inline test/extended/builds/multistage.go:26]: Not sure if this matters, but the above line that was replaced did not have the `as other`

File: test/extended/builds/multistage.go
Comment (coreydaley) [inline test/extended/builds/multistage.go:84]: The line that this replaced does not have the `as other`

File: test/extended/builds/multistage.go
Comment (gabemontero) [reply test/extended/builds/multistage.go:26]: Yeah that was intentional @coreydaley ... in the 4.7 changes the `as other` was added by Clatyon

**Review by coreydaley — COMMENTED** [review]
Just the question about the couple of "as other" usages that did not previously exist.

File: test/extended/builds/multistage.go
Comment (adambkaplan) [reply test/extended/builds/multistage.go:26]: `as other` is an alias in a Dockerfile. ok to keep for a multi-stage build test.

Comment (coreydaley) [reply test/extended/builds/multistage.go:26]: got it, thanks for the clarification.

**Review by adambkaplan — APPROVED** [review]
/lgtm

File: test/extended/builds/multistage.go
Comment (gabemontero) [reply test/extended/builds/multistage.go:26]: For reference the 4.7 flavor:  https://github.com/openshift/origin/blob/4cbb844d1d5f3a3ac6dd08325db17b807e7129f2/test/extended/builds/multistage.go#L27

**Comment by gabemontero** [comment]
/hold 

the e2e-aws-upgrade failure around `LEASED_RESOURCE: unbound variable" is showing up in a lot of places according to https://search.ci.openshift.org/?search=LEASED_RESOURCE%3A+unbound+variable&maxAge=24h&context=2&type=all&name=&maxMatches=5&maxBytes=20971520&groupBy=job

I've reached out on #4-dev-triage for BZ routing guidance

**Comment by gabemontero** [comment]
> /hold
> 
> the e2e-aws-upgrade failure around `LEASED_RESOURCE: unbound variable" is showing up in a lot of places according to https://search.ci.openshift.org/?search=LEASED_RESOURCE%3A+unbound+variable&maxAge=24h&context=2&type=all&name=&maxMatches=5&maxBytes=20971520&groupBy=job
> 
> I've reached out on #4-dev-triage for BZ routing guidance

Looks like we need https://github.com/openshift/ci-tools/pull/1535, minimally

**Comment by abhat** [comment]
@gabemontero there are still some 4.6 vsphere and vsphere-upi jobs failing with the docker rate limiting issue.

https://search.ci.openshift.org/?search=%5C%5Bsig-network%5C%5D+multicast+when+using+one+of+the+plugins+%27redhat%2Fopenshift-ovs-multitenant%2C+redhat%2Fopenshift-ovs-networkpolicy%27+should+allow+multicast+traffic+in+namespaces+where+it+is+enabled&maxAge=168h&context=1&type=junit&name=4.6&maxMatches=5&maxBytes=20971520&groupBy=job

Are there more pending PRs that need to get in?

**Comment by abhat** [comment]
Example job that is failing: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ocp-4.6-e2e-vsphere/1346094150457495552

**Comment by gabemontero** [comment]
hey @abhat - yeah  for 4.6 I only handled the subset of e2e for the build component that Clayton took on with his recent changes in 4.7

I have no further PRs planned.

And I don't have the bandwidth and am not taking on any throttling related updates in 4.6 for the other sig-* e2e's in the way I addressed builds, much less trying to take on backporting Clayton's entire change.  I saw in the recent email from him that a 4.6.z cherrypick of his changes were being considered, and I saw an attempt in his PR to use the bot, but I hit conflicts and failed.

For example, in looking at https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ocp-4.6-e2e-vsphere/1346094150457495552 I see sig-imageregistry, sig-cli, sig-network, sig-apps but not sig-builds

IMO 4.6.z BZs should be opened against each of those sig-* components for any throttling that they are hitting.

**Pull request merged**
